language-design.md

```diff
@@ -106,7 +209,25 @@ you do so? If not, why not?
+it is simply better for the task. However, when new ideas come along, to begin with, the ideas are explained using natural language and older notation. The optimal descriptions for learning concepts are not the same as the optimal descriptions for using them.
 
+This mirrors the tradeoffs in language design pretty well. Ultimately, notation specially crafted for the task really is better, and not just for efficiency's sake. The price is the learning curve. There is also the risk of a syntax not becoming widely adopted and therefore not being suitable for discussing ideas with other people. The idea of extensible languages is somewhat of a solution to this. New concepts are first described in terms of primitives people already understand, but the universal language is flexible enough to allow new and more specialized notation to be added in beyond that point. This is how mathematics itself handles the problem.
+
+The priority should probably be on how well the language design fits the actual task, on how good it is for people to use rather than to learn. Within that realm, you should optimize for familiarity. This means that, where suitable domain-specific notation and concepts exist, they should be used. This includes natural language, so if there is a natural language way to write something that doesn't really have any drawbacks, it should be preferred. Thus, for example, you should prefer Haskell's "if .. then .. else .." to Java's ".. ? .. : ..".
+
+There are two final interesting ideas that might also be useful. First, as mentioned before, extensible languages naturally have a way to solve this problem to some extent, as useful new notation will simply appear as it becomes useful. Given that, it may be best to not only aim for *extensibility*, but also *reducibility*. If a language appears like it could have been an extension of some simpler language, then it may be easier to learn. Additionally, they will still be useful to people who only know a subset of the language.
+
+Second, one last extremely hypothetical thought. Even in extensible languages, there is still the question of when to use the less convenient but more primitive description, and when to use the specialized notation. Looking up mathematical ideas on Wikipedia is awful because of how much terminology gets used. For someone who already understands the terminology, it's the best way to write it. But, if a more primitive description could be given without making it too much larger, then arguably that's the better explanation. IDEs may, in the future, provide some solution to this problem. As long as the mapping between primitive and specialized notation is one-to-one enough, it might be possible for two people to view and write the exact same code, but in different modes better suited to their understanding.
```

Comment:
I like this idea of an IDE that allows people to write and collaborate on the same code but in separate languages so that people can work with the language "better suited to their understanding."